adagrams/game.py

```diff
@@ -1,11 +1,125 @@
+import random
+
 def draw_letters():
-    pass
+    letters = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+    hand = []
+    for i in range(10):
+        choose_letter = list(letters.keys())
```

Comment:
Consider renaming `choose_letter` to `letter_pool` or maybe `complete_letters_list`. We want our names to describe what the variable references and the datatype too if it's helpful.

Taking a closer look at what letters.keys() returns, I see that it returns a list of 26 letters A-Z. This doesn't quite reflect the distribution of letters that your `letters` dictionary shows. How could you iterate over letters to create a list with 9 As, 2 Bs, 2 Cs, etc..?

Reply:
This is a great idea. This was the other place where I was confused by the feedback, but thinking about it again, if the dictionary is supposed to be a constant, this list would be another way to draw letters without modifying the dictionary if it contained the number of each letter.

Looking at it, I realized I also made a mistake because I should have had that list outside of the for loop. I don't need to recreate the list for each iteration, I want it to remain constant. Considering your feedback above, I wanted it to be it be a constant, so the variable name should have been in all capital letters, and this would have made my code easier to read. I also know there is probably an easier way to do this without creating a separate list and just using the dictionary keys, but in the moment, it felt like it made the code easier for me to read, write, and think about with a separate list.

What I wanted my code to do was not to remove the letters from the list, but use the list as a constant and randomly select a letter from that list for each draw in the hand. Then, when a letter was randomly selected, I wanted my loop to check to make sure that its value in the dictionary was still above zero (and to pick a different letter if the value was zero) and then subtract one from the value of the valid letter choice to show that one is now in the hand and no longer in the pool.

I see now that I could have done it the way that you're suggesting, and that would have made the letters dictionary a constant and the list of letters a variable. Is there a benefit to doing it that way, or was my code just confusing to read?

Reply:
The rest of the logic on lines 36-40 look good to me. I was just referring to the list choose_letter. How could your dict `letters` be leveraged to make a list with ALL the tiles?

What about something like:
```python
choose_letter = []
for letter in letters:
        choose_letter.extend([letter for i in range(letters[letter])])
```